### PR TITLE
Coverage QC report 

### PR DESCRIPTION
**Pull request by hassanfa**
This PR addresses #2328 

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN

### CODE REVIEW

**Comment by hassanfa** [comment]
Test 1: Scout stage mongo db. files exist:
image

Test 2: scout upload from CLI and exists on frontend:
image

Test 3: Report is accessible:
image

File: scout/commands/load/report.py [hunk 1]
```diff
@@ -54,3 +54,26 @@ def cnv_report(case_id, report_path, update):
     except Exception as err:
         LOG.error(err)
         raise click.Abort()
+
+
+@click.command("coverage-qc-report")
```
Comment (dnil) [inline scout/commands/load/report.py:59]: 👍🏻

File: scout/adapter/mongo/case.py [hunk 2]
```diff
@@ -702,6 +702,7 @@ def update_case(self, case_obj, keep_date=False):
             - madeline_info: If there is a new pedigree
             - vcf_files: paths to the new files
             - cnv_report: path to the CNV report file
+            - coverage_qc_report: path to the coverage and qc report file
```
Comment (dnil) [inline scout/adapter/mongo/case.py:705]:
```suggestion
            - coverage_qc_report: path to the static coverage and qc report file
```

File: scout/demo/cancer_coverage_qc_report.html [hunk 3]
```diff
@@ -0,0 +1,307 @@
+<!DOCTYPE html>
+<!--Report template and solution from: https://github.com/NationalGenomicsInfrastructure/ngi_reports-->
```
Comment (dnil) [inline scout/demo/cancer_coverage_qc_report.html:2]: Super with an example! NGI? Does this particular one have a history to document? 😊

File: scout/demo/cancer_coverage_qc_report.html [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+<html>
+<head>
+  <meta charset="utf-8">
+  <meta name="generator" content="Markdown">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
+  <title> klinisk sekvensering av cancer prover -  Kvalitetsrapport</title>
```
Comment (dnil) [inline scout/demo/cancer_coverage_qc_report.html:8]:
```suggestion
  <title> Klinisk sekvensering av cancerprover -  Kvalitetsrapport</title>
```

Comment (dnil) [reply scout/demo/cancer_coverage_qc_report.html:8]: Pardon my Swedish?

File: scout/demo/cancer_coverage_qc_report.html [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+    text-align:right;
+    margin-left:228px;
+  }
+
+</style>
+</head>
+<body>
+
+<h1 class="title">Kvalitetsrapport</h1>
+<h1 class="subtitle">klinisk sekvensering av cancer prover</h1>
```
Comment (dnil) [inline scout/demo/cancer_coverage_qc_report.html:218]:
```suggestion
<h1 class="subtitle">Klinisk sekvensering av cancerprover</h1>
```

Comment (dnil) [reply scout/demo/cancer_coverage_qc_report.html:218]: Ja..

File: scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html [hunk 6]
```diff
@@ -465,3 +468,13 @@ <h5 class="modal-title" id="archiveCaseLabel">Confirm archive case</h5>
   </div>
 </div>
 {% endmacro %}
+
+{% macro coverage_qc_report(institute, case) %}
+<div href="#" class="bg-dark list-group-item text-white">
+  <div class="d-flex flex-row flex-fill bd-highlight"></span>
+    <span class="menu-collapsed ml-2">Coverage and QC report
+      <a href="{{ url_for('cases.coverage_qc_report', institute_id=institute,
+                        case_name=case.display_name, format='html') }}" target="_blank"><i class="fa fa-file-pdf-o"></i></a>
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/collapsible_actionbar.html:477]: Hm, so here you might want to do two things: 
- use fa-link for the html link
- in contrast to the CNV report, also add a pdf generation link like for the case delivery report

File: scout/server/blueprints/cases/views.py [hunk 7]
```diff
@@ -872,6 +872,17 @@ def cnv_report(institute_id, case_name):
...
 
+@cases_bp.route("/<institute_id>/<case_name>/coverage-qc-report")
+def coverage_qc_report(institute_id, case_name):
+    """Display coverage and qc report."""
+    data = controllers.multiqc(store, institute_id, case_name)
+    if data["case"].get("coverage_qc_report") is None:
+        return abort(404)
+    out_dir = os.path.abspath(os.path.dirname(data["case"]["coverage_qc_report"]))
+    filename = os.path.basename(data["case"]["coverage_qc_report"])
+    return send_from_directory(out_dir, filename)
```
Comment (dnil) [inline scout/server/blueprints/cases/views.py:883]: ..and same here then: add in a if format=="pdf" render_pdf(HTML(source)) - see nice generalisable code a few lines above.. 😸

File: tests/load/test_load_report.py [hunk 8]
```diff
@@ -87,3 +87,27 @@ def test_load_cnv_report_using_case_id_with_update_success(adapter, cancer_case_
     # THEN a report should have been added to that case
     updated_case_obj = adapter.case_collection.find_one()
     assert updated_case_obj["cnv_report"] == report_path
+
+
+def test_load_coverage_qc_report_using_case_id_with_update_success(adapter, cancer_case_obj):
```
Comment (dnil) [inline tests/load/test_load_report.py:92]: 👍

**Review by dnil — CHANGES_REQUESTED** [review]
Super @hassanfa - really nicely done! I had to find something: have a look at the option to render the html file to pdf on the fly. It works really well for the delivery and general report, also it is much appreciated over on the hospital side where they often save a pdf report from into LIMS to step through different stages - at least one for qc, one for analysis.

File: scout/demo/cancer_coverage_qc_report.html (hunk 3 above)
Comment (hassanfa) [reply scout/demo/cancer_coverage_qc_report.html:2]: It is from this PR on BALSAMIC: https://github.com/Clinical-Genomics/BALSAMIC/pull/543

And template (CSS, table format, etc) taken from https://github.com/NationalGenomicsInfrastructure/ngi_reports/blob/master/data/report_templates/project_summary.html with ok from authors. :-)

**Comment by hassanfa** [comment]
I hope I got it right. Most of it copy-paste from working examples (which are pretty well written and I could follow).

**Comment by hassanfa** [comment]
Any idea why actions are not going forward?

**Comment by northwestwitch** [comment]
> Any idea why actions are not going forward?

Weird! Not really

**Comment by hassanfa** [comment]
I'll make a tiny commit to see if it triggers it.

**Comment by hassanfa** [comment]
Seems like it worked.

**Comment by northwestwitch** [comment]
> I'll make a tiny commit to see if it triggers it.

One thing that is happening is that some of my comments on another PR are not loaded, it says pending. Might be github!

**Comment by hassanfa** [comment]
PDF and HTML link on stage:
image

**Comment by northwestwitch** [comment]
How is it going with this one? When we merge it we could release a new version of Scout.. Today?

**Comment by dnil** [comment]
I'm sure it's fine, but I'll have a look. Agreed with new version! Did you want to try that primary transcript thing first as well, or would you rather we leave that for another week?

**Comment by northwestwitch** [comment]
> I'm sure it's fine, but I'll have a look. Agreed with new version! Did you want to try that primary transcript thing first as well, or would you rather we leave that for another week?

Let's leave it. I don't have much time to work on it today

**Review by dnil — APPROVED** [review]
Works, looks great!

**Pull request merged**